eth/blockwatch/client.go

```diff
@@ -0,0 +1,129 @@
+package blockwatch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// Client defines the methods needed to satisfy the client expected when
+// instantiating a Watcher instance.
+type Client interface {
+	HeaderByNumber(number *big.Int) (*MiniHeader, error)
+	HeaderByHash(hash common.Hash) (*MiniHeader, error)
+	FilterLogs(q ethereum.FilterQuery) ([]types.Log, error)
+}
+
+// RPCClient is a Client for fetching Ethereum blocks from a specific JSON-RPC endpoint.
+type RPCClient struct {
+	rpcClient      *rpc.Client
+	client         *ethclient.Client
+	requestTimeout time.Duration
+}
+
+// NewRPCClient returns a new Client for fetching Ethereum blocks using the given
+// ethclient.Client.
+func NewRPCClient(rpcURL string, requestTimeout time.Duration) (*RPCClient, error) {
+	ethClient, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	rpcClient, err := rpc.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	return &RPCClient{rpcClient: rpcClient, client: ethClient, requestTimeout: requestTimeout}, nil
+}
+
+type getBlockByNumberResponse struct {
+	Hash       common.Hash `json:"hash"`
+	ParentHash common.Hash `json:"parentHash"`
+	Number     string      `json:"number"`
+}
+
+// MiniHeader is a succinct representation of an Ethereum block header
+type MiniHeader struct {
```

Comment:
We can remove this type definition since `MiniHeader` was included in #1030

Reply:
fixed in 1861b5e